repository/git.go

```diff
@@ -177,6 +178,27 @@ func (r *Repository) propagateToWorktree(ctx context.Context, env *environment.E
 	return nil
 }
 
+func (r *Repository) exportEnvironment(ctx context.Context, env *environment.Environment) error {
+	worktreePointer := fmt.Sprintf("gitdir: %s/worktrees/%s", r.forkRepoPath, env.ID)
+
+	_, err := env.Workdir().
+		WithNewFile(".git", worktreePointer).
```

Comment:
micro nit: I think this is the first time we're crossing the boundary of exposing dagger outside of `environment`.

Couldn't we achieve the same with a dumb `os.WriteFile` here and leaving `Export()` as it is?

Reply:
I don't think it is, otherwise there would be new imports? Will check in a bit.

Reply:
yeah, we're also using dagger in Repository.Create, that boundary was broken when i did "real container resumes". 

and frankly i think that boundary is kinda necessary because we use dagger to manipulate host disk and I don't want environment to know about host disk.